semanticcpg/src/main/scala/io/shiftleft/semanticcpg/language/nodemethods/AstNodeMethods.scala

```diff
@@ -18,4 +18,19 @@ class AstNodeMethods(val node: nodes.AstNode) extends AnyVal {
     * */
   def astMinusRoot: AstNode = node.start.astMinusRoot
 
+  def isControlStructure: Boolean = node.start.isControlStructure.size == 1
+
+  def depth(p: nodes.AstNode => Boolean = _ => true): Int = {
+
```

Comment:
```suggestion
    node.start.astChildren.map(_.depth(p)).l.fold(if (p(node)) { 1 } else { 0 })(Math.max)
```
At least the last line can be a fold instead.

Reply:
As discussed offline, leaving it as-is for now.